src/compiler/checker.ts

```diff
@@ -16326,7 +16328,7 @@ namespace ts {
             }
 
             if (entityName.parent.kind === SyntaxKind.ExportAssignment) {
-                return resolveEntityName(<Identifier>entityName,
+                return resolveEntityName(<Identifier><EntityName>entityName,

```

Comment:
Yes, that extra cast was a left over from a work around for the earlier issue with the `isDeclarationName` user defined type guard.